extensions/ql-vscode/src/view/common/DataGrid.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from "react";
+import { forwardRef } from "react";
+import { styled } from "styled-components";
+
+/*
+ * A drop-in replacement for the VSCodeDataGrid family of components.
+ *
+ * The difference is that the `display: grid` styling is applied to `DataGrid`, whereas
+ * in the VS Code version that styling is applied to `VSCodeDataGridRow`. This gives
+ * column alignment across rows in situation with dynamic contents. It also allows
+ * for cells to span multiple rows and all the other features of data grids.
+ */
+
+const StyledDataGrid = styled.div<{ $gridTemplateColumns: string | number }>`
+  display: grid;
+  grid-template-columns: ${(props) => props.$gridTemplateColumns};
+  box-sizing: border-box;
+  width: 100%;
+  background: transparent;
+`;
+
+interface DataGridProps {
+  gridTemplateColumns: string | number;
+  children: React.ReactNode;
+}
+
+export function DataGrid({ gridTemplateColumns, children }: DataGridProps) {
+  return (
+    <StyledDataGrid $gridTemplateColumns={gridTemplateColumns}>
+      {children}
+    </StyledDataGrid>
+  );
+}
+
+const StyledDataGridRow = styled.div<{ $focused?: boolean }>`
+  display: contents;
+
+  &:hover > * {
+    background-color: var(--list-hover-background);
+  }
+
+  & > * {
+    // Use !important to override the background color set by the hover state
+    background-color: ${(props) =>
+      props.$focused
+        ? "var(--vscode-editor-selectionBackground) !important"
+        : "inherit"};
+  }
+`;
+
+interface DataGridRowProps {
+  focused?: boolean;
+  children: React.ReactNode;
+  "data-testid"?: string;
+}
+
+export const DataGridRow = forwardRef(
+  (props: DataGridRowProps, ref?: React.Ref<HTMLElement | undefined>) => {
```

Comment:
You should be able to do something like this here to avoid the `props["data-testid"]`:

```suggestion
  ({ focused, children, "data-testid": testId }: DataGridRowProps, ref?: React.Ref<HTMLElement | undefined>) => {
```